bpf/kmesh/probes/metrics.h

```diff
@@ -34,28 +36,42 @@ struct {
     __uint(max_entries, RINGBUF_SIZE);
 } map_of_metric_notify SEC(".maps");
 
-static inline void construct_metric_key(struct bpf_sock *sk, struct metric_key *key)
+static inline void construct_metric_key(struct bpf_sock *sk, __u8 direction, struct metric_key *key)
 {
     bpf_memset(key, 0, sizeof(struct metric_key));
-    if (sk->family == AF_INET) {
-        key->src_ip.ip4 = sk->src_ip4;
-        key->dst_ip.ip4 = sk->dst_ip4;
+
+    key->direction = direction;
+    if (direction == OUTBOUND) {
+        if (sk->family == AF_INET) {
+            key->src_ip.ip4 = sk->src_ip4;
+            key->dst_ip.ip4 = sk->dst_ip4;
+        } else {
+            bpf_memcpy(key->src_ip.ip6, sk->src_ip6, IPV6_ADDR_LEN);
+            bpf_memcpy(key->dst_ip.ip6, sk->dst_ip6, IPV6_ADDR_LEN);
+        }
+        key->dst_port = bpf_ntohl(sk->dst_port);
     } else {
-        bpf_memcpy(key->src_ip.ip6, sk->src_ip6, IPV6_ADDR_LEN);
-        bpf_memcpy(key->dst_ip.ip6, sk->dst_ip6, IPV6_ADDR_LEN);
+        if (sk->family == AF_INET) {
+            key->src_ip.ip4 = sk->dst_ip4;
+            key->dst_ip.ip4 = sk->src_ip4;
+        } else {
+            bpf_memcpy(key->src_ip.ip6, sk->dst_ip6, IPV6_ADDR_LEN);
+            bpf_memcpy(key->dst_ip.ip6, sk->src_ip6, IPV6_ADDR_LEN);
+        }
+        key->dst_port = sk->src_port;
```

Comment:
shouldn't  this be dst_port>

Reply:
When DIRECTION is INBOUND, the src and dst of the `sk` are the opposite of the src and dst of the traffic. 
So when direction is inbound, `dst_port` is `src_port`